src/System.Management.Automation/engine/lang/scriptblock.cs

```diff
@@ -896,44 +892,46 @@ private static Collection<PSObject> GetWrappedResult(List<object> result)
-                                           object scriptThis,
-                                           object[] args)
+        internal object DoInvokeReturnAsIs(
+            bool useLocalScope,
+            ErrorHandlingBehavior errorHandlingBehavior,
+            object dollarUnder,
+            object input,
+            object scriptThis,
+            object[] args)
         {
             List<object> result = new List<object>();
             Pipe outputPipe = new Pipe(result);
-            InvokeWithPipe(useLocalScope: useLocalScope,
-                           errorHandlingBehavior: errorHandlingBehavior,
-                           dollarUnder: dollarUnder,
-                           input: input,
-                           scriptThis: scriptThis,
-                           outputPipe: outputPipe,
-                           invocationInfo: null,
-                           args: args);
+            InvokeWithPipe(
+                useLocalScope: useLocalScope,
+                errorHandlingBehavior: errorHandlingBehavior,
+                dollarUnder: dollarUnder,
+                input: input,
+                scriptThis: scriptThis,
+                outputPipe: outputPipe,
+                invocationInfo: null,
+                args: args);
             return GetRawResult(result);
         }
 
         internal void InvokeWithPipe(
-                                    bool useLocalScope,
-                                    ErrorHandlingBehavior errorHandlingBehavior,
-                                    object dollarUnder,
-                                    object input,
-                                    object scriptThis,
-                                    Pipe outputPipe,
-                                    InvocationInfo invocationInfo,
-                                    bool propagateAllExceptionsToTop = false,
-                                    List<PSVariable> variablesToDefine = null,
-                                    Dictionary<string, ScriptBlock> functionsToDefine = null,
-                                    object[] args = null)
+            bool useLocalScope,
+            ErrorHandlingBehavior errorHandlingBehavior,
+            object dollarUnder,
+            object input,
+            object scriptThis,
+            Pipe outputPipe,
+            InvocationInfo invocationInfo,
+            bool propagateAllExceptionsToTop = false,
+            List<PSVariable> variablesToDefine = null,
+            Dictionary<string, ScriptBlock> functionsToDefine = null,
+            object[] args = null)
         {
             bool shouldGenerateEvent = false;
             bool oldPropagateExceptions = false;
             ExecutionContext context = LocalPipeline.GetExecutionContextFromTLS();
 
-            if (SessionStateInternal != null && SessionStateInternal.ExecutionContext != context)
+            if (SessionStateInternal?.ExecutionContext != context)
```

Comment:
@iSazonov this was the problematic change. It _looks_ harmless enough, but I guess something is slightly screwy with how this is evaluated when we change it in this way.

Reply:
Ah, of course-- that first conditional is being inverted here. Silly me. 😄

Reply:
Yes, it looks tricky.
I think it would be better to revert all such patterns.

Reply:
Yeah, you might be right. There's one where it's actually just doing a `value != null && value.property != null` which should be fine to leave, but the rest... better safe than sorry. 😄